modules/dnn/src/dnn.cpp

```diff
@@ -3205,15 +3227,51 @@ Net Net::Impl::createNetworkFromModelOptimizer(InferenceEngine::CNNNetwork& ieNe
 #ifdef HAVE_DNN_NGRAPH
         if (DNN_BACKEND_INFERENCE_ENGINE_NGRAPH == getInferenceEngineBackendTypeParam())
         {
+            const auto& outputName = it.first;
             Ptr<Layer> cvLayer(new NgraphBackendLayer(ieNet));
+            cvLayer->name = outputName;
+            cvLayer->type = "_unknown_";
 
-            InferenceEngine::CNNLayerPtr ieLayer = ieNet.getLayerByName(it.first.c_str());
-            CV_Assert(ieLayer);
+            if (ngraphFunction)
+            {
+                CV_TRACE_REGION("ngraph_function");
+                bool found = false;
+                for (const auto op : ngraphOperations)
+                {
+                    if (op->get_friendly_name() == outputName)
+                    {
+                        const std::string typeName = op->get_type_info().name;
+                        cvLayer->type = typeName;
+                        if (typeName == "DetectionOutput")
+                        {
+                            auto detOutput = std::dynamic_pointer_cast<ngraph::op::DetectionOutput>(op);
+                            if (detOutput)
+                            {
+                                int num_classes = detOutput->get_attrs().num_classes;
+                                ld.params.set("num_classes", num_classes);
+                                CV_LOG_INFO(NULL, "DNN/IE: num_classes=" << num_classes << " (" << outputName << " @ " << typeName << ")");
+                            }
+                            else
+                            {
+                                CV_LOG_WARNING(NULL, "DNN/IE: Can't extract layer information: '" << outputName << "'");
+                            }
+                        }
```

Comment:
@dkurt Do we need something like this? (extraction of some parameters)

Reply:
Not sure that we need this. DetectionOutput returns just indices of detected classes. Do we use the number of classes somewhere in the tests?

Reply:
Unused functionality is removed.